utils/helper.js

```diff
@@ -238,4 +240,63 @@ module.exports = {
       message: `User ${userName} uploaded using ${libraryName}`,
     });
   },
+  downloadFile: async (downloadUrl, localFilepath) => {
+    try {
+      const fileRes = await fetch(downloadUrl, {
+        method: "GET",
+        headers: new Headers({
+          "Content-Type": "application/pdf",
+        }),
+      });
+      const fileBuffer = await fileRes.buffer();
+      await fs.promises.writeFile(localFilepath, fileBuffer);
+      return {
+        writeFileStatus: 200,
+      };
+    } catch (error) {
+      logger.log({
+        level: "error",
+        message: `downloadFile: ${error}`,
+      });
+      return error;
+    }
+  },
+  uploadToCommons: async (metadata) => {
+    try {
+      const bot = await Mwn.init({
+        apiUrl: "https://commons.wikimedia.org/w/api.php",
+        OAuth2AccessToken: metadata.oauthToken,
+        userAgent: "bub2.toolforge ([[https://bub2.toolforge.org]])",
+        defaultParams: {
+          assert: "user",
+        },
+      });
+
+      const commonsFilePayload = "commonsFilePayload.pdf";
+      const title = metadata.details.volumeInfo.title || metadata.name;
+      const response = await bot.upload(
+        commonsFilePayload,
+        title,
+        metadata.commonsMetadata
+      );
+      if (await response.filename) {
+        await fs.promises.unlink(commonsFilePayload);
+      }
+      logger.log({
+        level: "info",
+        message: `Polling - uploadToCommons: Upload of ${metadata.IAIdentifier} to commons successful`,
```

Comment:
Is the word _Polling_ actually required here and below?

Reply:
@Parthiv-M I have made the requested changes